display_list/display_list_path_effect.cc

```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/display_list/display_list_path_effect.h"
+#include <memory>
+#include "include/core/SkRefCnt.h"
+
+namespace flutter {
+
+std::shared_ptr<DlPathEffect> DlPathEffect::From(SkPathEffect* sk_path_effect) {
+  if (sk_path_effect == nullptr) {
+    return nullptr;
+  }
+  // There are no inspection methods for SkPathEffect so we cannot break
+  // the Skia filter down into a specific subclass (i.e. DlSumPathEffect or
+  // DlComposePathEffect).
+  return std::make_shared<DlUnknownPathEffect>(sk_ref_sp(sk_path_effect));
+}
```

Comment:
PathEffect objects are not obtained from Flutter's dart:ui since they don't exist in that space.

They can only come from the text package (SkParagraph) and they are delivered to us through a Canvas adapter, so if we can't inspect them to turn them into a Dl object, then this wrapper is useless.

I think, though, that the DashPathEffect can be inspected and converted here so maybe this isn't a loss cause if that can be done.